frontend/app/components/api/api-v3/hal/hal-resource.service.ts

```diff
@@ -85,16 +86,22 @@ function halResource(halTransform, HalLink, $q) {
     }
 
     public $load() {
-      if (!this.$loaded) {
-        return this.$links.self().then(resource => {
-          this.$loaded = true;
+      if (this.$loaded) {
+        return $q.when(this);
+      }
 
-          angular.extend(this, resource);
-          return this;
-        });
+      if (!this.$loaded && this._loadingPromise) {
+        return this._loadingPromise;
       }
 
-      return $q.when(this);
+      this._loadingPromise = this.$links.self().then(resource => {
```

Comment:
While the schema is not $loaded, calling `$load` multiple will cause multiple requests / transforms to be made. If you return the first promise instead, it is saving all those requests.